src/ddprof_module_lib.cc

```diff
@@ -15,12 +15,33 @@
 #include <fcntl.h>
 #include <gelf.h>
 #include <libelf.h>
-#include <sys/stat.h>
-#include <sys/types.h>
 #include <unistd.h>
 
 namespace ddprof {
 
+class FileDescriptorHolder {
+public:
+  FileDescriptorHolder() : _fd(-1) {}
+  DDRes open_file(std::string_view path) {
+    _fd = ::open(path.data(), O_RDONLY);
+    if (_fd < 0) {
+      LG_WRN("[Mod] Couldn't open fd to module (%s)", path.data());
+      return ddres_warn(DD_WHAT_MODULE);
+    }
+    LG_DBG("[Mod] Success opening %s, ", path.data());
+    return ddres_init();
+  }
+
+  void take_ownership() { _fd = -1; }
```

Comment:
I would follow `unique_ptr` API and name this `release` instead, because here I would expect for `take_ownership` that `FileDescriptorHolder` takes the ownership of the file descriptor